Subject: Default value of usbhid.pb_fnmode module parameter
From: olh@suse.de
References: 220266

 The kernel default value for usbhid.pb_fnmode is 1, which means that pressing
 the Fn keys (F1..F10) without the fn key triggers the special functions
 decrease/increase brightness, mute, decrease/increase volume, etc., which is
 the default under MacOS.

 At least under 10.2 Beta2, only the volume related special functions work at
 all. In addition, Ctrl-Alt-Fx is used to switch between consoles. with
 pb_fnmode==1, the fn key needs to be pressed in addition.

 Therefore, pb_fnmode==2 (F1..F10 by default trigger Fn rather than the special
 functions) makes more sense under Linux.


 drivers/hid/hid-input.c |    2 +-
 1 files changed, 1 insertion(+), 1 deletion(-)

--- linux-2.6.20.orig/drivers/hid/hid-input.c
+++ linux-2.6.20/drivers/hid/hid-input.c
@@ -35,7 +35,7 @@
 
 #include <linux/hid.h>
 
-static int hid_pb_fnmode = 1;
+static int hid_pb_fnmode = 2;
 module_param_named(pb_fnmode, hid_pb_fnmode, int, 0644);
 MODULE_PARM_DESC(pb_fnmode,
 		"Mode of fn key on PowerBooks (0 = disabled, 1 = fkeyslast, 2 = fkeysfirst)");
